src/modules/processing/services/vimeo.js

```diff
@@ -1,108 +1,92 @@
+    let quality = obj.quality === "max" ? 9000 : Number(obj.quality);
+    if (quality < 240) quality = 240;
+    if (!quality || obj.isAudioOnly) quality = 9000;
 
     const url = new URL(`https://player.vimeo.com/video/${obj.id}/config`);
     if (obj.password) {
         url.searchParams.set('h', obj.password);
     }
 
-    let api = await fetch(url)
+    const api = await fetch(url)
                     .then(r => r.json())
                     .catch(() => {});
     if (!api) return { error: 'ErrorCouldntFetch' };
 
-    let downloadType = "dash";
-
-    if (!obj.isAudioOnly && JSON.stringify(api).includes('"progressive":[{'))
-        downloadType = "progressive";
-
-    let fileMetadata = {
+    const fileMetadata = {
         title: cleanString(api.video.title.trim()),
         artist: cleanString(api.video.owner.name.trim()),
     }
 
-    if (downloadType !== "dash") {
-        if (qualityMatch[quality]) quality = qualityMatch[quality];
-        let all = api.request.files.progressive.sort((a, b) => Number(b.width) - Number(a.width));
-        let best = all[0];
-
-        let bestQuality = all[0].quality.split('p')[0];
-        if (qualityMatch[bestQuality]) {
-            bestQuality = qualityMatch[bestQuality]
-        }
-
-        if (Number(quality) < Number(bestQuality)) {
-            best = all.find(i => i.quality.split('p')[0] === quality);
-        }
-
-        if (!best) return { error: 'ErrorEmptyDownload' };
-
-        return {
-            urls: best.url,
-            audioFilename: `vimeo_${obj.id}_audio`,
-            filename: `vimeo_${obj.id}_${best.width}x${best.height}.mp4`
-        }
-    }
-
     if (api.video.duration > env.durationLimit)
         return { error: ['ErrorLengthLimit', env.durationLimit / 60] };
 
-    let masterJSONURL = api.request.files.dash.cdns.akfire_interconnect_quic.url;
-    let masterJSON = await fetch(masterJSONURL).then(r => r.json()).catch(() => {});
+    const urlMasterHLS = api.request.files.hls.cdns.akfire_interconnect_quic.url;
 
-    if (!masterJSON) return { error: 'ErrorCouldntFetch' };
-    if (!masterJSON.video) return { error: 'ErrorEmptyDownload' };
+    const masterHLS = await fetch(urlMasterHLS)
+                            .then(r => r.text())
+                            .catch(() => {});
 
-    let masterJSON_Video = masterJSON.video
-                            .sort((a, b) => Number(b.width) - Number(a.width))
-                            .filter(a => ["dash", "mp42"].includes(a.format));
+    if (!masterHLS) return { error: 'ErrorCouldntFetch' };
 
-    let bestVideo = masterJSON_Video[0];
-    if (Number(quality) < Number(resolutionMatch[bestVideo.width])) {
-        bestVideo = masterJSON_Video.find(i => resolutionMatch[i.width] === quality)
-    }
+    const variants = HLS.parse(masterHLS).variants.sort(
+        (a, b) => Number(b.bandwidth) - Number(a.bandwidth)
+    );
+    if (!variants) return { error: 'ErrorEmptyDownload' };
```

Comment:
variants can exist and be a non-empty array, so line 56 would throw an exception